R/create_query_data.R

```diff
@@ -395,35 +331,37 @@ get_terms_from_db <- function(version,
                               expect_query_name = FALSE,
                               expect_query_id = FALSE,
                               i,
-                              temp_env,
-                              type) {
+                              temp_env) {
   assert_db_requirements(
     version = version,
     version_arg_name = arg_name(substitute(version)),
     fun = fun,
     fun_arg_name = arg_name(substitute(fun)),
     queries = queries,
-    i = i,
-    type = type
+    i = i
   )
-  select <- rlang::parse_expr(paste0(str_to_lower(type), "_select = definition"))
   fun_call <- quo(fun(
-    !!select,
+    basket_select = definition,
     version = version,
     keep_id = expect_query_id,
-    temp_env = temp_env
+    temp_env = temp_env,
+    type = definition$type
```

Comment:
This line should be removed. As the type is included in the `basket_select` object, we do not need a separate argument for it.